### DISCUSSION TITLE
PHP frontend deprecation and Python performance

### DISCUSSION THREAD

**Discussion by Roze** [osm-search/Nominatim#3560] [General]
I haven't tested with the recent 4.5 release, but for 4.4 I tried to evaluate the python frontend. 

The generic benchmark for our test dataset yielded something like:

```
Concurrency Level:      100
Complete requests:      5000
```

PHP:
```
Requests per second:    434.02 [#/sec] (mean)
Time per request:       2.304 [ms] (mean, across all concurrent requests)
```
vs Python (both falcon and starlette were similar) 
```
Requests per second:    177.20 [#/sec] (mean)
Time per request:       5.643 [ms] (mean, across all concurrent requests)
```
Quite a significant performance drop.

Which is also in line with the https://nominatim.org/2023/04/03/reverse-performance.html Nominatim author self-tests. .

Has there been improvements? 
If not then losing php would be somewhat sad.

**Comment by lonvia** [comment]
There are no significant changes for the reverse endpoint between the PHP and Python implementation except for the addition of the layer parameter. The important improvements happened in the search endpoint.

The PHP frontend will not come back. You can maintain your own fork of the frontend and use it with the standard database. There is no guarantee though that the database schema will remain compatible forever.

**Comment by Roze** [comment]
But the reason for dropping is just extra burden of maintenance (lack of developers or interest) or are there some technical nuances?

**Reply by lonvia** [reply]
The PHP code has aged to degree that it has become unmaintainable. For more details on the reasoning behind the switch please read [this blog post](https://nominatim.org/2022/12/08/frontend-renovation.html).
